### PR TITLE
A handy date to duration converter

### PR DESCRIPTION
**Pull request by shana**
This will be used in the pr features.

### CODE REVIEW

**Comment by Haacked** [comment]
What's with the Resource designer files? Were they intended to be part of this PR?

**Comment by shana** [comment]
@Haacked They're for future eventual localization of the strings of the conversion. "x days ago", "x minutes ago", etc

**Comment by Haacked** [comment]
Ah, cool! I'm currently reviewing.

File: src/GitHub.UI/Converters/DurationToStringConverter.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            const int month = 30;
+            const int day = 24;
+            const int hour = 60;
+            const int minute = 60;
+
+            TimeSpan duration;
+            if (value is TimeSpan)
+                duration = (TimeSpan)value;
+            else if (value is DateTime)
+                duration = DateTime.Now - (DateTime)value;
```
Comment (Haacked) [inline src/GitHub.UI/Converters/DurationToStringConverter.cs]: Where would we be getting `value` from. Is it possible it'd be stored as `UTC` and we should use `UtcNow` here? Or can we assume it's always local time?

File: src/GitHub.UI/Converters/DurationToStringConverter.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Globalization;
+
+namespace GitHub.UI
+{
+    public class DurationToStringConverter : ValueConverterMarkupExtension<DurationToStringConverter>
+    {
+        public override object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            const int year = 356;
```
Comment (Haacked) [inline src/GitHub.UI/Converters/DurationToStringConverter.cs]: Did you mean 365? I assume it just needs to be close so we can ignore leap years. :smile:

Comment (Haacked) [inline src/GitHub.UI/Converters/DurationToStringConverter.cs]: Should  this be 365?

**Comment by Haacked** [comment]
:pineapple:

File: src/GitHub.UI/Converters/DurationToStringConverter.cs (hunk 2 above)
Comment (shana) [inline src/GitHub.UI/Converters/DurationToStringConverter.cs]: lol, how the heck did that happen and obviously I'm not testing hard enough!

File: src/GitHub.UI/Converters/DurationToStringConverter.cs (hunk 1 above)
Comment (shana) [inline src/GitHub.UI/Converters/DurationToStringConverter.cs]: Good question... data can come from any number of sources, most often from the github api, so it's going to be UTC. /me fixicates

**Comment by shana** [comment]
:truck:

**Comment by Haacked** [comment]
:sparkles:

**Pull request merged**